### PR TITLE
Refunds and Nonce

### PR DESCRIPTION
**Pull request by anthonyvscode**

### CODE REVIEW

File: src/Pinch.SDK/Payments/PaymentClient.cs [hunk 1]
```diff
@@ -233,5 +233,16 @@ public async Task<ApiResponse> Delete(string id)
...
         }
+
+        /// <summary>
+        /// Check a payment nonce
+        /// </summary>
+        /// <param name="options">Payment nonce.</param>
+        /// <returns></returns>
+        public async Task<ApiResponse<PaymentDetailed, PaymentCheckNonceOptions>> CheckNonce(PaymentCheckNonceOptions options)
+        {
+            var response = await PostHttp<PaymentDetailed, PaymentCheckNonceOptions>("payments/nonce", options);
```
Comment (bjcull) [inline src/Pinch.SDK/Payments/PaymentClient.cs:244]: Do we need the options included in the response? Usually only for options without IDs

File: src/Pinch.SDK/Pinch.SDK.csproj [hunk 2]
```diff
@@ -11,9 +11,9 @@
     <PackageId>Pinch.SDK</PackageId>
     <PackageTags>payments;pinch;sdk;direct debit</PackageTags>
     <PackageIconUrl>https://raw.githubusercontent.com/bjcull/Pinch.SDK/master/assets/img/circle_logo_64x64.png</PackageIconUrl>
```
Comment (bjcull) [inline src/Pinch.SDK/Pinch.SDK.csproj:13]: does this reference need the org changed too?

File: src/Pinch.SDK/Refunds/RefundsClient.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        /// <summary>
+        /// Create a Refund
+        /// </summary>
+        /// <param name="options">Refund information.</param>
+        /// <returns></returns>
+        public async Task<ApiResponse<Refund, RefundSaveOptions>> Save(RefundSaveOptions options)
+        {
+            var response = await PostHttp<Refund, RefundSaveOptions>("refunds", options);
```
Comment (bjcull) [inline src/Pinch.SDK/Refunds/RefundsClient.cs:27]: Ditto for options in the repsonse

File: src/Pinch.SDK/Refunds/RefundsClient.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        /// <summary>
+        /// Check a refund nonce
+        /// </summary>
+        /// <param name="options">Refund nonce.</param>
+        /// <returns></returns>
+        public async Task<ApiResponse<Refund, RefundCheckNonceOptions>> CheckNonce(RefundCheckNonceOptions options)
+        {
+            var response = await PostHttp<Refund, RefundCheckNonceOptions>("refunds/nonce", options);
```
Comment (bjcull) [inline src/Pinch.SDK/Refunds/RefundsClient.cs:38]: options in response

**Review by bjcull — CHANGES_REQUESTED** [review]
Check comments and let me know

**Comment by bjcull** [comment]
Also, be sure to push this one as a minor pre-release. I'd suggest 0.18.1-refunds1

File: src/Pinch.SDK/Pinch.SDK.csproj (hunk 2 above)
Comment (anthonyvscode) [reply src/Pinch.SDK/Pinch.SDK.csproj]: fixed in https://github.com/PinchPayments/Pinch.SDK/commit/719414d82e93f2e70de356376ff6ff911be577b2

File: src/Pinch.SDK/Payments/PaymentClient.cs (hunk 1 above)
Comment (anthonyvscode) [reply src/Pinch.SDK/Payments/PaymentClient.cs]: Removed https://github.com/PinchPayments/Pinch.SDK/commit/88e772f17df5169b96ecea39b6b79f80a2c9395e

File: src/Pinch.SDK/Refunds/RefundsClient.cs (hunk 4 above)
Comment (anthonyvscode) [reply src/Pinch.SDK/Refunds/RefundsClient.cs]: Removed https://github.com/PinchPayments/Pinch.SDK/commit/b6343a5fd0ae2db96000d85cae91d68c27964840

File: src/Pinch.SDK/Refunds/RefundsClient.cs (hunk 3 above)
Comment (anthonyvscode) [reply src/Pinch.SDK/Refunds/RefundsClient.cs]: Removed https://github.com/PinchPayments/Pinch.SDK/commit/b6343a5fd0ae2db96000d85cae91d68c27964840

**Review by bjcull — APPROVED** [review]
Cool, looks good

**Pull request merged**